gk/fib.c

```diff
@@ -221,7 +221,7 @@ parse_ip_prefix(const char *ip_prefix, struct ipaddr *res)
 	if (ip_prefix == NULL)
 		return -1;
 
-	strncpy(ip_prefix_copy, ip_prefix, ip_prefix_len + 1);
+	memcpy(ip_prefix_copy, ip_prefix, ip_prefix_len + 1);
```

Comment:
The size of the array `ip_prefix_copy`  is guaranteed to be exactly what is needed to copy `ip_prefix`, so why not replace `strncpy()` with `strcpy()`?

If it works, the same solution can be applied to the other case.

Reply:
By the way, the issue is better explained [here](https://stackoverflow.com/questions/56782248/gcc-specified-bound-depends-on-the-length-of-the-source-argument).

Remember to update the description of the patch.